service/key/key.go

```diff
@@ -267,15 +269,31 @@ func RemoteWritePasswordKey() string {
 // We do not have a provider agnostic label like "giantswarm.io/version" to define this.
 func IsCAPICluster(obj metav1.Object) bool {
 	// TODO once we have migrated all clusters to CAPI, we can remove this
-	switch v := obj.(type) {
-	case *capiv1alpha3.Cluster:
-		if _, ok := v.Labels["azure-operator.giantswarm.io/version"]; ok {
+
+	checker := func(labels map[string]string) bool {
```

Comment:
@whites11 is this code anywhere else ? Would it make sense to package that as a giantswarm lib ? Something like capi-utils ?

Reply:
I would not be sure where to put it honestly.
I agree it would be nice though.
Task for a real PE I guess :sweat_smile: